go.mod

```diff
@@ -261,4 +261,6 @@ require (
 	lukechampine.com/blake3 v1.1.7 // indirect
 )
 
+replace go.opencensus.io => go.opencensus.io v0.23.0
```

Comment:
Don't use replace rules it's dependency hell, just `go get go.opencensus.io@v0.23.0` or whatever.

Reply:
> a dependency of kubo depends on an old version of go.opencensus.io, so it is included no matter what. the only way to make sure old versions of opencensus are not included is to use a replace rule (or find the dependencies and fix those).

IIUC that's not how the Go module versioning system works. It's only capable of including code from a single major version. That means that if go.opencensus.io v0.23.0 is included in kubo the code from earlier v0 versions (such as v0.18.0 which includes the dependency on that apache commit) are not included in the code). Since kubo already uses v0.23.0  (see https://github.com/ipfs/kubo/blob/58aaee00f80cf913174cf47f8c40367b70b8a5df/go.mod#L101) there shouldn't be any problems here.


Code showing the dependency relying on thrift
```
❯ go mod graph | select-string "git.apache.org/thrift.git"
go.opencensus.io@v0.18.0 git.apache.org/thrift.git@v0.0.0-20180902110319-2566ecd5d999
```